js/reducers/index.js

```diff
@@ -63,34 +65,27 @@ function data(state = INITIAL_DATA, action) {
 
 function setAnonymous(state, anonymous) {
   let idx = 1;
-  const newStudents = state.get("students").map(s => s.set("name", anonymous ? `Student ${idx++}` : s.get("realName")));
-  return state.set("anonymous", anonymous)
-    .set("students", newStudents);
+  const newStudents = state.get("students")
+    .map(s => s.set("name", anonymous ? `Student ${idx++}` : s.get("realName")));
+  return state.set("anonymous", anonymous).set("students", newStudents);
 }
 
-// This action has to be explicit and requires additional property. Otherwise, a question will disappear immediately
-// when teacher has selection filter active and unselects given question. Instead, the question should stay visible
-// until teacher clicks "Show selected" again.
```

Comment:
Note that you lost this behavior described in this comment. Selecting a question immediately triggers filtering, which feels a bit strange, as other questions disappear and you cannot select them anymore. You can compare:
http://portal-report.concord.org/branch/166240880-assignment-level-report-settings/index.html
and
https://portal-report.concord.org/version/v1.7.1/index.html

Reply:
Ah, right!  I didn't immediately understand this.  

Thanks.

Reply:
I restored this functionality starting on line 89 below.

Reply:
👍